File: app/src/main/java/com/example/radioplayer/infrajava/RadioStationRepo.java
package com.example.radioplayer.infrajava;

import com.example.radioplayer.domainjava.RadioStationsResponse;

import retrofit2.Call;


/**
 * Radio station repo interface with methods for specific operations
 */
public interface RadioStationRepo {
    Call<RadioStationsResponse> getStationsByCountryCode(int country);

    Call<RadioStationsResponse> getStationByRPUID(String rpuid);
}

File: app/src/main/java/com/example/radioplayer/infra/RadioStationRepo.kt
package com.example.radioplayer.infra

import com.example.radioplayer.domain.RadioStationsResponse
import retrofit2.Response
import javax.inject.Inject

/** Radio station repo interface with methods for specific operations */
interface RadioStationRepo {
    suspend fun getStationsByCountryCode(
        country: Int
    ): Response<RadioStationsResponse>

    suspend fun getStationByRPUID(
        rpuid: String
    ): Response<RadioStationsResponse>
}

/**Implementation of Radio station repo class */
class RadioStationsRepoImpl @Inject constructor(private val radioStationService: RadioStationService) :
    RadioStationRepo {
    override suspend fun getStationsByCountryCode(
        country: Int
    ): Response<RadioStationsResponse> {
        return radioStationService.getStationsByCountryCode(country)
    }

    override suspend fun getStationByRPUID(
        rpuid: String
    ): Response<RadioStationsResponse> {
        return radioStationService.getRadioByRpuid(rpuid)
    }
}
